### ISSUE TITLE
Region Click Event is undefined.

### ISSUE THREAD

**Issue by blairjam** [manifestinteractive/jqvmap#166]
On line 467 of jquery.vmap.js, regionClickEvent is undefined.

This issue is solved by creating regionClickEvent a few lines above the if statement that checks against it.

```javascript
465.    var regionClickEvent = $.Event("regionClick.jqvmap");
466.    jQuery(params.container).trigger(regionClickEvent, [code, mapData.pathes[code].name]);
467.    if (!regionClickEvent.isDefaultPrevented()) {
468.        if (map.selectedRegions.indexOf(code) !== -1) {
469.            map.deselect(code, path);
470.        } else {
471.            map.select(code, path);
472.        }
473.    }
```

**Comment by billpull** [comment]
Any chance of this getting merged into an official build?

**Comment by blairjam** [comment]
Probably not. I don't even know if this project is being maintained anymore. The last commit is from over 2 years ago. Someone else would need to fork and maintain it.

**Comment by patricknelson** [comment]
Agreed on this. The code is going to suffer without something to maintain this. Maybe we can get the maintainer @peter-schmalfeldt to hand off access to this repository to someone else. Hopefully he's alive and well :smile:

**Comment by patricknelson** [comment]
And just in case this works, pinging @manifestinteractive as well. Is there anyone else who can maintain this project, since it's already so popular now and ranks really high on Google?

**Comment by manifestinteractive** [comment]
Anyone that wants to maintain this, let me know.  I put feelers out a while ago when I knew I was going to get slammed with freelance work, but did not get any takers.  I can add anyone to the contributors list that wants to help maintain this.

**Comment by manifestinteractive** [comment]
@patricknelson Thanks for the concern, I am indeed alive and well :)  I just got hit with three new freelance contracts back in July and things got crazy busy for me.  I can add you as a contributor to this project if you want to help maintain it, just let me know.

**Comment by patricknelson** [comment]
Hmm... I'm only at the research stage for a project coming up and I was testing some things out and immediately found a bug, looked at the repo and noticed all the open issues/PR's so I flagged you down.

To help find someone more likely to be dedicated (sorry I know nothing at all about the internals of this), might I recommend:

- Posting something on the website near the github/download icon?
- Post something in the README.md that people (devs, likely contributes) will see immediately?
- Follow up with people you've seen post lots of issues/PR's who you think will know more about the repo?

I'll dive deeper into this later (as I'm certain I'll end up using it) and I'll follow up here eventually when  I'm more certain about contributing. Chances are I'll put in a few PR's or want to merge some of the outstanding ones since the project may have some exposure.

**Comment by manifestinteractive** [comment]
@patricknelson I've tried contacting people directly as you suggested.  No one want to maintain it.  And with most of the issues ( like this one ) being code people want added, without making a Pull request, it becomes a nightmare to maintain.  The reason that most of the issues are not resolved and PR's not merged is because the issues are people posting code fixes in comments, rather than actually integrating it into the code and submitting a pull request.  As you'll also see, the PR's we do have, conflict with the master branch, meaning people that submitted the PR's did not use the latest master branch before submitting their PR.

So, what's left is a HUGE amount of work to sort out the literally hundreds of issues and PR's that can't be automatically merged because someone either asked for a feature as an issue ( without a PR, like this ticket ) or created a PR that can't be merged.

To sort all this out means someone would have to start from the beginning and manually work on integrating / rejecting everything that was committed as either an issue or PR.  Which is a few weeks of work for anyone that wants to take it on.

If I get a gap in my freelance work, I will look into taking this head on, but in the mean time, I am hoping that those asking for me to help, would be willing to help as well.  That's kind of the point of Free Open Source Software.  It's not supposed to be one person doing all the maintenance, but I will keep looking for people that are willing to help keep this project alive.

In regards to this issue, if you submit the fix as a pull request, and can verify it is working with Master ( meaning the PR will LET me merge it in without merge conflicts ) I can easily get this integrated today.  But as it is only a comment in a ticket, there is nothing I can do without a PR.

**Comment by patricknelson** [comment]
Trust me I understand that; I don't have any issues yet. I've done a lot of work on some other open source projects. Basically if I were to take it over I'd want to ensure I had a process for everything, including code contribution. 

*I'd have some ground rules:*

1. Ensure you've rebased
1. Squash your commits into a single commit
1. Always include the FIX or NEW keywords along with the associated issue number formatted like "FIX #1: My message here"
1. Create a separate branch. 

Fail to meet those guidelines, or don't contribute usable code and it'll be rejected. As for existing PR's you'd have to take it as they come. Aside from that however I'm interested in the *licensing* if I were to try to maintain it, as I'd be concerned of legal issues (since I'd be interested in using THIS in a commercial product). 

- Does this have a large amount of code copied from jVectorMap? They have licensing that's not entirely free and that is free is GNU, not MIT (like this project). Commercial use must be paid. 
- Where do you source the actual map data (pretty core to future maintenance/updates)? Is it from here http://jvectormap.com/maps/world/south-america/ or is it more directly from their source, Natural Earth Vector? Their licensing is permissive enough to match up with your current licensing. Basically we'd want to setup a process to integrate future maps and updates to existing maps by either coding out a script to convert their raw data, write a document outlining manual steps or simply pull from jVectorMaps and hope they don't mind...

Those are my only two main concerns.

**Comment by manifestinteractive** [comment]
This project was initially a deviation from jVectorMap, so much so I did not anticipate the original author wanting to to integrate the changes in as it was a pretty big rewrite to support the new features I needed.  I still keep reference to the original project for those that might want to use it instead.

In regards to licensing, my project was started before jVectorMap added the licenses he has now.  Meaning that as long as you are using only my project, you can use the [license](https://github.com/manifestinteractive/jqvmap/blob/stable/LICENSE) I have on file, which was added in 2013.  jVectorMap [Commercial license](https://github.com/bjornd/jvectormap/commits/master/LICENSE-COMMERCIAL) was not added until Oct 1, 2014.  Since I have not ported over any code from jVectorMap after 2013, you can use my library with my existing licenses.

The Map data, as mentioned above, comes from their 2013 repository, before any licensing was added to their project. Other maps have been provided through pull requests from other project contributors.

So, this project was only started using the public code jVectorMap released in 2013.  I have made an effort to not cross any paths with that project since they added commercial licensing requirements.

Hope that answers any questions you may have.

**Comment by patricknelson** [comment]
Yep :) I know it's frustrating but maybe this thread will serve as informative for others!

I just noticed map data can be converted directly from the original source, sorry about that distraction. Out of curiosity what features does jqvmap have that jVectorMap does not? I was not able to deduce very much from commit history.

**Comment by manifestinteractive** [comment]
It was ported two years ago, I have no idea what the differences are anymore.  I know the creator kept working on new features, just like I had people contribute new featured to this one.

**Comment by manifestinteractive** [comment]
OK, I have done a massive update to the codebase to both resolve the open pull requests ( all but 6 of the 26 were able to be merged ).  This issue is resolved in v1.1.0.  For those following the thread, I have also added a CONTRIBUTING.md file that outlines how to contribute to this project.
